benchmarks.ts

```diff
@@ -46,6 +49,58 @@ function measure<T=void>(message: (t:T) => string, callback: () => T, minMillise
   return result;
 }
 
+function intersectBySorting(
+  tree1: BTree<number, number>,
+  tree2: BTree<number, number>,
+  callback: (k: number, leftValue: number, rightValue: number) => void
+) {
+  const left = tree1.toArray();
+  const right = tree2.toArray();
+  let i = 0;
+  let j = 0;
+  const leftLen = left.length;
+  const rightLen = right.length;
+
+  while (i < leftLen && j < rightLen) {
+    const [leftKey, leftValue] = left[i];
+    const [rightKey, rightValue] = right[j];
+    if (leftKey === rightKey) {
+      callback(leftKey, leftValue, rightValue);
+      i++;
+      j++;
+    } else if (leftKey < rightKey) {
+      i++;
+    } else {
+      j++;
+    }
+  }
+}
+
+function subtractBySorting(
```

Comment:
Add comment `/** calls `callback` for each key and value that is in `tree1` but not `tree2` (O(n)) */`